Wikipedia/UI-V5/WMFSaveButtonController.m

```diff
@@ -0,0 +1,83 @@
+//  Created by Monte Hurd on 8/19/15.2//  Copyright (c) 2015 Wikimedia Foundation. Provided under MIT-style license; please copy and modify!
+
+#import "WMFSaveButtonController.h"
+#import "MWKSavedPageList.h"
+#import "MWKArticle.h"
+#import "SessionSingleton.h"
+#import "MWKUserDataStore.h"
+
+@interface WMFSaveButtonController ()
+
+@property (strong, nonatomic) UIButton* button;
+
+@end
+
+@implementation WMFSaveButtonController
+
+- (instancetype)initWithButton:(UIButton*)button
+                         title:(MWKTitle*)title {
+    self = [super init];
+    if (self) {
+        self.button = button;
+        self.title  = title;
+        [self observeSavedPages];
+        [self updateSavedButtonState];
+    }
+    return self;
+}
+
+- (MWKSavedPageList*)savedPageList {
+    return [SessionSingleton sharedInstance].userDataStore.savedPageList;
```

Comment:
A central saved page list needs to be observed so changes are reflected anywhere this controller is used.